pepr/attack_runner.py

```diff
@@ -0,0 +1,208 @@
+        logger.info(f"Parse attack runner configuration file: {yaml_path}")
+        data = yaml.load(f_stream)
+        attack_obj_save_path
+
+        os.makedirs(attack_obj_save_path, exist_ok=True)
+        os.makedirs(report_save_path + "/fig", exist_ok=True)
+        sections = []
+
+        for i, yaml_attack_pars in enumerate(data["attack_pars"]):
+            attack_type = yaml_attack_pars["attack_type"]
+            logger.info(
+                f"Attack number: {i+1}\n\n"
+                "######################## Attack Run ########################"
+            )
+            if attack_type == "mia":
+                # Parse MIA configuration
+                logger.info("Setup configuration.")
+                shadow_model_function = functions[
+                    yaml_attack_pars["create_compile_shadow_model"]
+                ]
+                attack_model_function = functions[
+                    yaml_attack_pars["create_compile_attack_model"]
+                ]
+
+                yaml_data_conf = data["data_conf"][i]
+                shadow_indices = list(np.load(yaml_data_conf["shadow_indices_path"]))
+                target_indices = list(np.load(yaml_data_conf["target_indices_path"]))
+                evaluation_indices = list(
+                    np.load(yaml_data_conf["evaluation_indices_path"])
+                )
+                record_indices_per_target = np.load(
+                    yaml_data_conf["record_indices_per_target_path"]
+                )
+
+                target_models = []
+                for path in yaml_attack_pars["target_model_paths"]:
+                    target_models.append(tf.keras.models.load_model(path))
+
+                # Setup dictionaries for attack
+                attack_pars = {
+                    "number_classes": yaml_attack_pars["number_classes"],
+                    "number_shadow_models": yaml_attack_pars["number_shadow_models"],
+                    "shadow_training_set_size": yaml_attack_pars[
+                        "shadow_training_set_size"
+                    ],
+                    "create_compile_shadow_model": shadow_model_function,
+                    "create_compile_attack_model": attack_model_function,
+                    "shadow_epochs": yaml_attack_pars["shadow_epochs"],
+                    "shadow_batch_size": yaml_attack_pars["shadow_batch_size"],
+                    "attack_epochs": yaml_attack_pars["attack_epochs"],
+                    "attack_batch_size": yaml_attack_pars["attack_batch_size"],
+                }
+
+                data_conf = {
+                    "shadow_indices": shadow_indices,
+                    "target_indices": target_indices,
+                    "evaluation_indices": evaluation_indices,
+                    "record_indices_per_target": record_indices_per_target,
+                }
+
+                # Create attack object
+                logger.info(f"Attack: {yaml_attack_pars['attack_alias']}")
```

Comment:
Should we change this to "Attack alias: ..."?

Reply:
Okay, changed in ff16d50